dwave/system/composites/reversecomposite.py

```diff
@@ -0,0 +1,205 @@
+                                     **parameters)
+
+            initial_state, _ = dimod.as_samples(initial_state)
+            vectors = _update_data_vector(vectors, sampleset,
+                                          {'initial_state': [initial_state[0]] * len(sampleset.record.energy),
+                                           'schedule_index': [schedule_idx] * len(sampleset.record.energy)})
+
+            if parameters['reinitialize_state']:
+                # if reinitialize is on, choose the lowest energy, most probable state for next iteration
+                ground_state_energy = sampleset.first.energy
+                lowest_energy_samples = sampleset.record[sampleset.record.energy == ground_state_energy]
+                lowest_energy_samples.sort(order='num_occurrences')
+                initial_state = dict(zip(sampleset.variables, lowest_energy_samples[-1].sample))
+            else:
+                # if not reinitialized, take the last state as the next initial state
+                initial_state = dict(zip(sampleset.variables, sampleset.record.sample[-1]))
+
+        samples = vectors.pop('sample')
+        return dimod.SampleSet.from_samples((samples, bqm.variables),
+                                            bqm.vartype,
+                                            info={'anneal_schedules': anneal_schedules},
+                                            **vectors)
+
+
+class ReverseBatchStatesComposite(dimod.ComposedSampler):
+    """Composite that reverse anneals from multiple initial samples. Each submission is independent
+    from one another.
+
+    Args:
+       sampler (:obj:`dimod.Sampler`):
+            A dimod sampler.
+
+    """
+
+    def __init__(self, child_sampler):
+        self._children = [child_sampler]
+
+    @property
+    def children(self):
+        return self._children
+
+    @property
+    def parameters(self):
+        param = self.child.parameters.copy()
+        param['initial_states'] = []
+        return param
+
+    @property
+    def properties(self):
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample(self, bqm, **parameters):
+        """Sample the binary quadratic model using reverse annealing from multiple initial states.
+
+        Args:
+            bqm (:obj:`dimod.BinaryQuadraticModel`):
+                Binary quadratic model to be sampled from.
+
+            **parameters:
+                Parameters for the sampling method, specified by the child sampler.
+
+        Returns:
+            :obj:`dimod.SampleSet` that has initial_state field.
+
+        """
+        child = self.child
+
+        if 'initial_states' not in parameters:
+            return child.sample(bqm, **parameters)
+
+        initial_states = parameters.pop('initial_states')
+
+        # there is gonna be way too much data generated - better to histogram them
+        parameters['answer_mode'] = 'histogram'
```

Comment:
This should check if `answer_mode` is an available parameter of the child sampler